lambdas/service/app.py

```diff
@@ -780,7 +780,7 @@ def file_url(uuid, **params):
 
 def self_url(endpoint_path=None):
     protocol = app.current_request.headers.get('x-forwarded-proto', 'http')
-    base_url = app.current_request.headers['host']
+    base_url = app.current_request.headers.get('host')
     if endpoint_path is None:
         endpoint_path = app.current_request.context['path']
     retry_url = f'{protocol}://{base_url}{endpoint_path}'
```

Comment:
… produce a URL that reads `http://None/…` and causes more insidious failures later.